mongo/src/main/java/com/redhat/lightblue/mongo/crud/MongoCRUDController.java

```diff
@@ -715,15 +715,16 @@ private void createUpdateEntityInfoIndexes(EntityInfo ei, Metadata md) {
                 LOGGER.info("Creating index {} with {}",index.getName(),index.getFields());
                 DBObject newIndex = new BasicDBObject();
                 for (IndexSortKey p : index.getFields()) {
-                    String field = Translator.translatePath(p.getField());
+                    String field = p.getField().toString();
```

Comment:
Declare this as Path field, get rid of toString(), things will simplify going forward